pkg/test/controller/reconciler/testreconciler.go

```diff
@@ -240,15 +238,8 @@ func (r *TestReconciler) newReconcilerForCRD(crd *apiextensions.CustomResourceDe
 		if crd.GetLabels()[k8s.DCL2CRDLabel] == "true" {
 			return dclcontroller.NewReconciler(r.mgr, crd, r.dclConverter, r.dclConfig, r.smLoader, immediateReconcileRequests, resourceWatcherRoutines, defaulters, jg)
 		}
-
-		switch crd.GetName() {
-		case "logginglogmetrics.logging.cnrm.cloud.google.com":
-			m, err := logging.GetModel(context.TODO(), &controller.Config{HTTPClient: r.httpClient})
-			if err != nil {
-				return nil, fmt.Errorf("error getting logging model: %w", err)
-			}
-
-			return directbase.NewReconciler(r.mgr, immediateReconcileRequests, resourceWatcherRoutines, loggingapis.LoggingLogMetricGVK, m, jg)
+		if directbase.ControllerBuilder.IsDirectByCRD(crd) {
+			return directbase.ControllerBuilder.NewReconciler(r.mgr, &kcccontroller.Config{HTTPClient: r.httpClient}, immediateReconcileRequests, resourceWatcherRoutines, crd.GroupVersionKind(), jg)
```

Comment:
Nit: IIRC crd.GroupVersionKind() is problematic, ideally ControllerBuilder would record the GVK